File: clients/JavaScript/my-strategy.js
const model = require('./model/index');

class MyStrategy {
    async getAction(playerView, debugInterface) {
        return new model.Action(new Map());
    }
    async debugUpdate(playerView, debugInterface) {
        await debugInterface.send(new model.DebugCommand.Clear());
        await debugInterface.getState();
    }
}

module.exports.MyStrategy = MyStrategy;

File: clients/Kotlin/src/main/kotlin/MyStrategy.kt
import model.*

class MyStrategy {
    fun getAction(playerView: PlayerView, debugInterface: DebugInterface?): Action {
        return Action(mutableMapOf())
    }
    fun debugUpdate(playerView: PlayerView, debugInterface: DebugInterface) {
        debugInterface.send(model.DebugCommand.Clear())
        debugInterface.getState()
    }
}
